load_eessi_extend_module.sh

```diff
@@ -0,0 +1,88 @@
+# Script to load the environment module for EESSI-extend.
+# If that module is not available yet, a specific version will be installed using the latest EasyBuild.
+#
+# This script must be sourced, since it makes changes in the current environment, like loading an EESSI-extend module.
+#
+# This script is part of the EESSI software layer, see
+# https://github.com/EESSI/software-layer.git
+#
+# author: Kenneth Hoste (@boegel, HPC-UGent)
+# author: Alan O'Cais (@ocaisa, CECAM)
+#
+# license: GPLv2
+#
+#
+set -o pipefail
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <EESSI-extend version>" >&2
+    exit 1
+fi
+
+EESSI_EXTEND_VERSION="${1}-easybuild"
+
+# make sure that environment variables that we expect to be set are indeed set
+if [ -z "${TMPDIR}" ]; then
+    echo "\$TMPDIR is not set" >&2
+    exit 2
+fi
+
+# ${EB} is used to specify which 'eb' command should be used;
+# can potentially be more than just 'eb', for example when using 'eb --optarch=GENERIC'
+if [ -z "${EB}" ]; then
+    echo "\$EB is not set" >&2
+    exit 2
+fi
+
+# make sure that utility functions are defined (cfr. scripts/utils.sh script in EESSI/software-layer repo)
+type check_exit_code
+if [ $? -ne 0 ]; then
+    echo "check_exit_code function is not defined" >&2
+    exit 3
+fi
+
+echo ">> Checking for EESSI-extend module..."
+
+ml_av_eessi_extend_out=${TMPDIR}/ml_av_eessi_extend.out
+module avail 2>&1 | grep -i EESSI-extend/${EESSI_EXTEND_VERSION} &> ${ml_av_eessi_extend_out}
+
+if [[ $? -eq 0 ]]; then
+    echo_green ">> Module for EESSI-extend/${EESSI_EXTEND_VERSION} found!"
+else
+    echo_yellow ">> No module yet for EESSI-extend/${EESSI_EXTEND_VERSION}, installing it..."
+
+    EB_TMPDIR=${TMPDIR}/ebtmp
+    echo ">> Using temporary installation of EasyBuild (in ${EB_TMPDIR})..."
+    pip_install_out=${TMPDIR}/pip_install.out
+    pip3 install --prefix ${EB_TMPDIR} easybuild &> ${pip_install_out}
+
+    # keep track of original $PATH and $PYTHONPATH values, so we can restore them
+    ORIG_PATH=${PATH}
+    ORIG_PYTHONPATH=${PYTHONPATH}
+
+    echo ">> Final installation in ${EASYBUILD_INSTALLPATH}..."
+    export PATH=${EB_TMPDIR}/bin:${PATH}
+    export PYTHONPATH=$(ls -d ${EB_TMPDIR}/lib/python*/site-packages):${PYTHONPATH}
+    eb_install_out=${TMPDIR}/eb_install.out
```

Comment:
Hmm, this looks odd to me. Should we not rely on having at least one EasyBuild module already in the stack? If there is none, then we should just stop the installation of the extend module.